packages/authorization-updater/src/index.ts

```diff
@@ -1,33 +1,126 @@
+/* eslint-disable functional/immutable-data */
+import { v4 as uuidv4 } from "uuid";
 import { runConsumer } from "kafka-iam-auth";
+import { match } from "ts-pattern";
 import { EachMessagePayload } from "kafkajs";
 import {
   messageDecoderSupplier,
   kafkaConsumerConfig,
   logger,
+  getContext,
 } from "pagopa-interop-commons";
+import {
+  Descriptor,
+  EServiceV2,
+  fromEServiceV2,
+  missingMessageDataError,
+  EServiceId,
+} from "pagopa-interop-models";
+import {
+  AuthorizationService,
+  authorizationServiceBuilder,
+} from "./authorization-service.js";
+
+const getDescriptorFromEvent = (msg: {
+  data: {
+    descriptorId: string;
+    eservice?: EServiceV2;
+  };
+}): {
+  eserviceId: EServiceId;
+  descriptor: Descriptor;
+} => {
+  if (!msg.data.eservice) {
+    throw missingMessageDataError("eservice", "EServiceDescriptorPublished");
+  }
+
+  const eservice = fromEServiceV2(msg.data.eservice);
+  const descriptor = eservice.descriptors.find(() => msg.data.descriptorId);
 
-async function processMessage({
-  topic,
-  message,
-  partition,
-}: EachMessagePayload): Promise<void> {
-  try {
-    const messageDecoder = messageDecoderSupplier(topic);
-    const decodedMessage = messageDecoder(message);
-
-    // TODO : update authorization to AuthjorizationManagement service
-
-    logger.info(
-      `Authorization updated after ${JSON.stringify(
-        decodedMessage.type
-      )} event - Partition number: ${partition} - Offset: ${message.offset}`
-    );
-  } catch (e) {
-    logger.error(
-      `Error during message handling. Partition number: ${partition}. Offset: ${message.offset}, ${e}`
-    );
+  if (!descriptor) {
+    throw missingMessageDataError("descriptor", "EServiceDescriptorPublished");
   }
```

Comment:
This always uses "EServiceDesccriptorPublished" as the second parameter but it's also called in the case of EServiceDescriptorActivated and EServiceDescriptorSuspended

Reply:
good catch!